### ISSUE TITLE
New component: SAP HANA Receiver

### ISSUE THREAD

**Issue by dehaansa** [open-telemetry/opentelemetry-collector-contrib#8827]
## The purpose and use-cases of the new component
The SAP HANA Receiver would collect metrics from SAP HANA Monitoring & Statistics views. It would leverage the golang [driver](https://github.com/SAP/go-hdb) written by SAP and the base sql package to execute the relevant queries. In this manner it could collect metrics about memory utilization, network usage, disk/volume IO, transactions, replication, and more.

It would target SAP HANA version 2 as older versions are out of maintenance at this time.

## Example configuration for the component
The following configuration settings are required:
- username
- password

The following configuration settings are optional:
- endpoint (default: localhost:30015): The host & port where SAP HANA can be reached
- collection_interval (default = 60s): This receiver collects metrics on an interval. Valid time units are ns, us (or µs), ms, s, m, h.
- tls (defaults defined [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)): TLS control. By default insecure settings are rejected and certificate verification is on.
```
receivers:
  saphana:
    endpoint: localhost:30015
    username: otelu
    password: $SAPHANA_PASSWORD
    collection_interval: 60s
```

## Telemetry data types supported
Metrics!

## Sponsor (Optional)
No sponsor at the moment.

**Comment by pmm-sumo** [comment]
I can volunteer as a sponsor :)

**Comment by djaglowski** [comment]
Closed by #9234
